server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -115,6 +116,8 @@ public final class RemoteSegmentStoreDirectory extends FilterDirectory implement
      */
     protected final AtomicBoolean canDeleteStaleCommits = new AtomicBoolean(true);
 
+    protected final AtomicBoolean isDirectoryCleanupOngoing = new AtomicBoolean(false);
```

Comment:
What protection does this boolean provide given that we create a new RemoteSegmentStoreDirectory on each invocation of `BlobStoreRepository::releaseRemoteStoreLockAndCleanup`?

Reply:
yes agree, its not helping much here since these directories are per shard, just using it in UT right now to wait for cleanup completion: [here](https://github.com/opensearch-project/OpenSearch/pull/12319/files#diff-0077a3751a4c11aaac36c9d3c07f4e874dcb719717a0509964671f85e4698cfcR952).